### PR TITLE
feat: add ruff-format option

### LINKED ISSUE #29: Feature: Add `ruff_format` option
See previous discussion in #27.

### PR DESCRIPTION
**Pull request by afuetterer**
Fixes #29

Tasks
- [x] add ruff_format CLI option
- [x] add tests
- [x] fix failing test

### CODE REVIEW

**Comment by afuetterer** [comment]
I am unsure how to hand over an argument in `self._post_transforms.append(self._run_ruff)`

So I opted for two methods.

File: test/test_tree_transformer.py [hunk 1]
```diff
@@ -1507,6 +1512,62 @@ def test_ruff_fix_pass_file_name(tmp_path, monkeypatch) -> None:
...
+
+    source = """
+    mode = "format"
+    """
+
+    expected = """
+    mode = 'format'
+    """
+
+    assert transformer(dedent(source)) == dedent(expected)
```
Comment (afuetterer) [inline test/test_tree_transformer.py:1536]: I feel kind of stupid. Why is this failing?

File: unasyncd/transformers.py [hunk 2]
```diff
@@ -179,19 +180,18 @@ def __init__(
...
-            "--no-cache",
-            "--fix",
-            "--fix-only",
-            "--quiet",
-        ]
+            self._post_transforms.append(self._ruff_fix)
+        if ruff_format:
+            self._post_transforms.append(self._ruff_format)
+
+    def _run_ruff(self, source: str, output: str, tree: cst.Module, mode: str) -> str:
```
Comment (afuetterer) [inline unasyncd/transformers.py]: Is this approach alright @provinzkraut?

**Comment by afuetterer** [comment]
Now I have introduced quite a lot of duplication in the tests. What do you suggest? Go for more parametrization?

ruff_fix and ruff_format should tested both individually and in combination, I guess?

File: test/test_tree_transformer.py (hunk 1 above)
Comment (provinzkraut) [reply test/test_tree_transformer.py:1536]: The block comments `"""` start with a newline. Ruff removes the leading newline. You'll need to make the code start on the same line as the comment to fix this :)

File: unasyncd/transformers.py (hunk 2 above)
Comment (provinzkraut) [reply unasyncd/transformers.py]: I'm not a huge fan of passing flags like this. I've left a different suggestion!

File: unasyncd/transformers.py [hunk 3]
```diff
@@ -179,19 +180,18 @@ def __init__(
...
+        args = [sys.executable, "-m", "ruff"]
+        if mode == "fix":
+            args.extend(["check", "--no-cache", "--fix", "--fix-only", "--quiet"])
+        elif mode == "format":
+            args.extend(["format", "--no-cache", "--quiet"])
+        else:
+            raise ValueError("Invalid mode.")
```
Comment (provinzkraut) [inline unasyncd/transformers.py]:
```suggestion
    def _run_ruff(self, source: str, output: str, tree: cst.Module, cmd: list[str]) -> str:
        args = [sys.executable, "-m", "ruff", *cmd, "--no-cache", "--quiet"]
```

File: unasyncd/transformers.py [hunk 4]
```diff
@@ -208,6 +208,12 @@ def _run_ruff(self, source: str, output: str, tree: cst.Module) -> str:
                 raise ChildProcessError(f"Error calling ruff: {stderr}")
             return stdout
 
+    def _ruff_fix(self, source: str, output: str, tree: cst.Module) -> str:
+        return self._run_ruff(source, output, tree, mode="fix")
+
+    def _ruff_format(self, source: str, output: str, tree: cst.Module) -> str:
+        return self._run_ruff(source, output, tree, mode="format")
+
```
Comment (provinzkraut) [inline unasyncd/transformers.py]:
```suggestion
    def _ruff_fix(self, source: str, output: str, tree: cst.Module) -> str:
        return self._run_ruff(source, output, tree, cmd=["check", "--fix", "--fix-only"])

    def _ruff_format(self, source: str, output: str, tree: cst.Module) -> str:
        return self._run_ruff(source, output, tree, cmd=["format"])
```
together with the changes proposed above

**Review by provinzkraut — CHANGES_REQUESTED** [review]

**Comment by provinzkraut** [comment]
> Now I have introduced quite a lot of duplication in the tests. What do you suggest? Go for more parametrization?

Not a lot of duplication really and having thing spelled out explicitly is never a bad idea :)

File: unasyncd/transformers.py (hunk 2 above)
Comment (afuetterer) [reply unasyncd/transformers.py]: Thank you!

File: test/test_tree_transformer.py (hunk 1 above)
Comment (afuetterer) [reply test/test_tree_transformer.py:1536]: Another one like this. This fails:

```python
source = """
import time, asyncio
mode = "format"
"""
```

But this as well:

```python
source = """import time, asyncio
mode = "format"
"""
```

Can you help me out again?

**Comment by afuetterer** [comment]
What do you think?

**Comment by provinzkraut** [comment]
> What do you think?

I have allowed myself to fix the test issue :)

Apart from that it looks good!

**Review by provinzkraut — APPROVED** [review]

**Comment by afuetterer** [comment]
Thanks a lot!

**Pull request merged**